providers/base/units/cpu/jobs.pxu

```diff
@@ -56,7 +56,7 @@ plugin: attachment
 category_id: com.canonical.plainbox::cpu
 id: after-suspend-cpu/scaling_test-log-attach
 estimated_duration: 1.0
-after: after-suspend-cpu/scaling_test
+depends: after-suspend-cpu/scaling_test
```

Comment:
`after` and `depends` do slightly different things:

- `after` means `after-suspend-cpu/scaling_test-log-attach` should be run after `after-suspend-cpu/scaling_test` regardless of its outcome. It's used so that Checkbox can still do something (for instance retrieve logs) ***even if*** `after-suspend-cpu/scaling_test` (the parent job) failed.
- if you use `depends`, `after-suspend-cpu/scaling_test-log-attach` will be skipped if `after-suspend-cpu/scaling_test` fails.

Now, what I'm not sure is why would this job (`after-suspend-cpu/scaling_test-log-attach`) fail in the first place. Its `command` field (below) says "if there is a log file at that location, attach it.". @weizhenwu would you have a sample submission where the problem you're trying to fix is exhibited?

Reply:
Here is the recent submission:
https://certification.canonical.com/hardware/202303-31378/submission/316037/test-results/?term=cpu%2Fscaling_test

the parent job `after-suspend-cpu/scaling_test` was skipped based on the resource false, so there is no log file existing to attach.
I used `depend` instead of `after` since it should be skipped as well if the parent job is skipped.